rdagent/scenarios/data_science/proposal/exp_gen/draft/draft.py

```diff
@@ -130,6 +130,31 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.supports_response_schema = APIBackend().supports_response_schema()
 
+        # Packages requested by the LLM during the draft stage. Cached for reuse.
+        self._requested_pkgs: list[str] | None = None
+
+    # ---------------------------------------------------------------------
+    # New: ask the LLM which third-party packages it plans to use so we can
+    # query their versions dynamically.
+    # ---------------------------------------------------------------------
+
+    def _package_query(self, scenario_desc: str) -> list[str]:
+        """Ask the LLM for the list of packages it intends to import.
+
+        Returns a unique, lowercase list of package names.
+        """
+        sys_prompt = T(".prompts_draft:pkg_query.system").r()
+        user_prompt = T(".prompts_draft:pkg_query.user").r(scenario_desc=scenario_desc)
+
+        response = APIBackend().build_messages_and_create_chat_completion(
```

Comment:
move this query prompt into rdagent/scenarios/data_science/proposal/exp_gen/proposal.py:L747